docs/content/direct/hub-chart.md

```diff
@@ -0,0 +1,157 @@
+# KubeStellar Hub chart usage
+
+This documents explains how to use KubeStellar Hub chart to deploy a new instance of KubeStellar
+with a choise of user-defined Control Plains.
+
+The information provided is specific for the following release:
+
+```shell
+export KUBESTELLAR_VERSION=0.23.0-alpha.1
+```
+
+## Pre-requisites
+
+A list of executable that may be required during the deployment can be fund [here](./pre-reqs.md).
+At minimum [Helm](https://helm.sh/) is required.
+
+The setup of KubStellar via the Hub chart requires the existance of at least one cluster
+to be used for the deployment of the chart.
+
+This can be:
+
+1. A local **Kind** cluster with an ingress with SSL passthrough and a mapping to host port 9443
+
+This option is particularly useful for first time users or users that would like to have a local deployment.
+
+For convenience, a new local Kind clusters that satisfies the requirements for KubeStellar setup
+and that can be used to exercises the [examples](./examples.md) can be created with the following command:
+
+```shell
+sudo su <<EOF
+bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${KUBESTELLAR_VERSION}/scripts/create-kind-cluster-with-SSL-passthrough.sh) --name kubestellar --port 9443
+EOF
+```
+
+It is important to note that, if a different name is used for the Kind cluster, then its control plane name must be also be referenced during the Helm chart installation by setting `--set "kubeflex-controller.hostContainer=<control-plane-name>".
+
+2. A local **k3s** cluster with an ingress with SSL passthrough and a mapping to host port 9443
+
+... Under onstruction ...
+
+3. An **OpenShift** cluster
+
+When using this option, one is required to explicitely set the `isOpenShift` variable to `true` by including `--set "kubeflex-operator.isOpenShift=true"` in the Helm chart installation command.
+
+
+## KubeStellar Hub Chart values
+
+The KubeStellar chart makes available to the user several values that may be used to customize its installation into an existing cluster:
+
+```yaml
+# KubeFlex override values
+kubeflex-operator:
+  install: true # enable/disable the installation of KubeFlex by the chart (default: true)
+  installPostgreSQL: true # enable/disable the installation of the appropriate version of PostgreSQL required by KubeFlex (default: true)
+  isOpenShift: false # set this variable to true when installing the chart in an OpenShift cluster (default: false)
+  # Kind cluster specific settings:
+  domain: localtest.me # used to set the Control Planes DNS in a Kind cluster installation (default: localtest.me)
+  externalPort: 9443 # used to set the port to access the Control Planes API (default: 944)
+  hostContainer: kubeflex-control-plane # used to set the name of cluster control plane (default: kubeflex-control-plane, which corresponds to a Kind cluster with name kubeflex)

```

Comment:
I do not understand some of the words here. What is "cluster control plane" trying to say? I am really lost trying to understand the parenthetic remark about default.

I am not sure whether this is relevant, but: when I do `kind create cluster --name kubeflex`, this creates a _context_ in my current kubeconfig file and this context is named "kind-kubeflex".

Reply:
It was not clear to me untill I stumbled into it.
When one does `kind create cluster --name foo`, the the cluster runs in a docker container called `foo-control-plane`
Then one must tell kflex to use that name `hostContainer: foo-control-plane`
In all our examples (expecially when `kflex init -c`) the kind cluster was called kubeflex... which is ok with all defaults.
I stambled into it because I thought that we may want to call the cluster kubestellar instead.

Please note that pending a new release of kflex, there is bug that actually prevent this.

Reply:
I understand now. See my comment on this entry in the values.yaml file.